eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -923,7 +926,7 @@ shelleyFixupTx =
     >=> addRootTxIn
     >=> fixupFees
     >=> updateAddrTxWits
-    >=> (\tx -> logFeeMismatch tx $> tx)
+    >=> (logFeeMismatch *> pure)
```

Comment:
This change is wrong, so let's bring back the original:
```suggestion
    >=> (\tx -> logFeeMismatch tx $> tx)
```

This is incorrect, because it will totally skip execution of `logFeeMismatch`, because composition of `logFeeMismatch` and `pure` with `*>` actually uses `Applicative` instance for `->`, not the underlying monad. It took me a moment to realize why it even compiles.

Here is an example of how it breaks:

```haskell
ghci> x <- (print *> pure) "foo"
ghci> x
"foo"
```

So, the lesson I learned today is that `*>` for functions simply ignores the first function:
```
1. f *> g
2. (id <$ f) <*> g
3. fmap (const id f) <*> g
4. fmap id <*> g
```

In the matter of fact, things like that is usually the reason why I try to stay away from esoteric instances like Functor/Applicative/Monad/.. for functions, Foldable for tuples, etc.

Reply:
Yikes! That's bad. Thanks for catching this, and for explaining. I'll be more cautious about using the `(->)` instances in future.

It makes sense that `*>` would skip the left argument, because in a pure context there's no point in executing something if you're not going to use the result.